dir.traversal.targets

```diff
@@ -6,7 +6,7 @@
     <!-- To Serialize we use msbuild's batching functionality '%' to force it to batch all similar projects with the same identity 
          however since the project names are unique it will essentially force each to run in its own batch -->
     <MSBuild Targets="Build" Projects="@(Project)" Condition="'$(SerializeProjects)'=='true'" Properties="Dummy=%(Identity)"/>
-    <MSBuild Targets="Build" Projects="@(Project)" Condition="'$(SerializeProjects)'!='true'" BuildInParallel="true" />
+    <MSBuild Targets="Build" Projects="@(Project)" Condition="'$(SerializeProjects)'!='true'" BuildInParallel="false" />
```

Comment:
The error happens when It is trying compile two projects "src\ToolBox\SOS\NETCore\SOS.NETCore.csproj" and "src\mscorlib\System.Private.CoreLib.csproj" in parallel. You can reproduce this issue using the following command "build.cmd checked x64 linuxmscorlib". Both of them could be compiled separately in any order when in parallel failing. So I think we have a "race condition" bug somewhere inside msbuild.

The error generally looks like this:
>C:\git\coreclr\Tools\versioning.targets(323,5): error MSB4062: The "GenerateCurrentVersion" task could not be loaded from the assembly C:\git\coreclr\Tools/Microsoft.DotNet.Build.Tasks.dll. Could not load file or assembly 'System.Runtime, Version=4.1.0.0, Culture=neutral, PublicKeyToken=b03f5f7f11d50a3a' or one of its dependencies. The system cannot find the file specified. Confirm that the <UsingTask> declaration is correct, that the assembly and all its dependencies are available, and that the task contains a public class that implements Microsoft.Build.Framework.ITask. [C:\git\coreclr\src\mscorlib\System.Private.CoreLib.csproj]

The exact target (GenerateCurrentVersion in my example) which could not be found depends on stars or something else. IMO the different projects redefine some internal data in msbuild due to multi-threading issue.